airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -78,7 +94,32 @@ def poll_job_completion(self):
                 raise Exception('%s has failed pods, failing task.' % self.job_name)
 
     def execute(self, context):
-        self.job_name = yaml.safe_load(self.job_yaml_string)['metadata']['name']
+        # create job_yaml_string from the parameter dictionary.
+        # append extra parameters to commands inside the job
+        if not self.job_yaml_string and self.job_param_dict and 'containers' in self.job_param_dict:
```

Comment:
I think that this logic should be pulled out into a util function, particularly since I don't think it will be the most common use case.

Ideally, the operator should only ever have to take a yaml string. It should be up to the developer to create that the right way (in this case, using an outside helper)

Reply:
I can move some of the logic out.

Is there any reason to only take a yaml string into operator?  This prevents us from creating command dynamically.  Particularly, xcom_pull has to be executed in an operator to pull the values.  The current API only takes static things.

Reply:
The command can be dynamically generated in the DAG if you use a PythonOperator. So something like:

def generate_command():
    # code to generate command

op = PythonOperator(task_id='generate-command-text',
                      python_callable=generate_command)

Reply:
I wanted to avoid putting logic into the Operator itself to generate the yaml because people will be able to do this in a variety of ways, otherwise I'm afraid this Operator will continue to keep expanding as people think of new yaml use cases

Reply:
I'm testing something out, will write back if it works

Reply:
we chatted: going to check this in and invest in more xcom stuff moving forward